eq-author/src/App/page/Design/Validation/DateValidation/index.js

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { flowRight, find } from "lodash";
+
+import { Grid, Column } from "components/Grid";
+import { ValidationPills } from "../ValidationPills";
+import ValidationTitle from "../ValidationTitle";
+import AlignedColumn from "../AlignedColumn";
+import Duration from "../Duration";
+import { ERR_NO_VALUE } from "constants/validationMessages";
+import withChangeUpdate from "enhancers/withChangeUpdate";
+import * as entityTypes from "constants/validation-entity-types";
+import { DATE } from "constants/answer-types";
+import { DAYS, MONTHS, YEARS } from "constants/durations";
+import PathEnd from "../path-end.svg?inline";
+
+import { StyledError, RelativePositionText, ConnectedPath, Now } from "./components";
+import PreviousAnswerEditor from "./PreviousAnswerEditor";
+import MetadataEditor from "./MetadataEditor";
+import CustomEditor from "./CustomEditor";
+import PositionPicker from "./PositionPicker.js";
+
+const START_COL_SIZE = 3;
+const END_COL_SIZE = 12 - START_COL_SIZE;
+const UNITS = [DAYS, MONTHS, YEARS];
+
+const getUnits = format => {
+  switch (format) {
+    case "dd/mm/yyyy":
+      return UNITS;
+    case "mm/yyyy":
+      return UNITS.slice(1);
+    default:
+      return UNITS.slice(2);
+  }
+}
```

Comment:
```suggestion

const getUnits = format => {
  switch (format) {
    case "dd/mm/yyyy":
      return [DAYS, MONTHS, YEARS];
    case "mm/yyyy":
      return [MONTHS, YEARS];
    default:
      return [YEARS];
  }
}
```

Just a bit of refactoring. You don't have to include it but I thought it might be a bit more readable.